src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/KinesisShardSyncer.java

```diff
@@ -79,17 +81,18 @@ synchronized void bootstrapShardLeases(IKinesisProxy kinesisProxy, ILeaseManager
      * @throws ProvisionedThroughputException
      * @throws KinesisClientLibIOException
      */
-    public synchronized void checkAndCreateLeasesForNewShards(IKinesisProxy kinesisProxy,
+    public synchronized void checkAndCreateLeasesForNewShards(List<Shard> shards, IKinesisProxy kinesisProxy,
```

Comment:
lets use `Optional<List<Shard>>`

Reply:
Seems using Optional is not recommended for Type of a field or a parameter, let's avoid using Optional here :)

Reply:
Why? It provides an elegant and robust way of handling nulls. Worth reading this - https://dzone.com/articles/optional-method-parameters

> There are some valid arguments against using Optional as method parameter type, but to me, they are not good enough to be able to say with confidence that you should avoid it at all costs. And don’t forget, there are also arguments for using Optional this way. I’d say that you should keep each of them in the back of your head and evaluate things on a case by case basis.